### ISSUE TITLE
Appending jobs dynamically onto an existing workflow

### ISSUE THREAD

**Issue by stevebauman** [ksassnowski/venture#45]
I'm wondering if it may be possible to append jobs onto an existing (and possibly in-progress) workflow, similarly to how [Laravel Bus Batches](https://laravel.com/docs/9.x/queues#adding-jobs-to-batches) allows you to append jobs onto a batch via the `$this->batch()->add($jobs)` method inside of jobs inside the batch:

```php
/**
 * Execute the job.
 *
 * @return void
 */
public function handle()
{
    if ($this->batch()->cancelled()) {
        return;
    }
 
    $this->batch()->add(Collection::times(1000, function () {
        return new ImportContacts;
    }));
}
```

Maybe an API like:

```php
use WorkflowStep;

public function handle()
{
    // Append a job...
    $this->workflow()->addJob(new AnotherJob);

    // Append a workflow...
    $this->workflow()->addWorkflow(new AnotherWorkflow);
}
```

I can work on a PR for this, but I wanted to gather any thoughts you may have and any gotchas that I may not be seeing?

Thanks for your time! 🙏

**Comment by ksassnowski** [comment]
Do you have an example of a possible use case for this?

**Comment by stevebauman** [comment]
Yea absolutely. At my work we're using AWS' Document Classification engine (ML) to classify text via a queued job. We must use a queued job because Amazon doesn't perform Document Classification instantly. Instead, we have to check in with Amazon to determine if the classification job is done, and then retrieve the response.

Once Amazon says it's done classifying the document, we pull out the response. With the responses' classification, we then determine if the text needs to go through any further classifications.

Below is a psuedo-code example and what I'd like to achieve:

```php
public function handle(Amazon $amazon)
{
    $job = $amazon->findJobById($this->amazonJobId);

    if (! $job->isDone()) {
        // Release the job to check again in 60 seconds.
        return $this->release(60);
    }

    $response = $job->getResponse();

    switch ($response->getClassification()) {
        case 'CATEGORY_1':
            return $this->workflow()->addJob(
                new DocumentClassfication($amazon->createJob('some-ml-model'))
            );
        case 'CATEGORY_2':
            return $this->workflow()->addJob(
                new DocumentClassfication($amazon->createJob('some-other-ml-model'))
            );
        case 'CATEGORY_3':
            return;
        default:
            throw new UnexpectedValueException();
    }
}
```

Basically, we need to be able to dynamically generate workflow tree's depending on the response from Amazon.

Thoughts?
